add example struct

oliverpool opened the pull request:
Related to #9

This replaces the `type Methods int` with a `type Struct struct{...}` to show how exported and unexported fields are shown on the documentation.

oliverpool committed:
add example struct

fluhus commented:
Awesome, thanks!

A few thoughts I had:
1. I think methods and structs are different things so I'd like them to appear under separate sections. Can you keep the `Methods` part as is, and add the Struct as its own thing?
2. The hidden field will be invisible in the rendered godoc. How about adding a copy of the struct code to the godoc itself, to demonstrate how it gets rendered? Like in the [examples](https://pkg.go.dev/github.com/fluhus/godoc-tricks#Examples) section. Something like:
   ```go
   // Something about structs...
   //
   //   type Struct struct {
   //     // fields...
   //   }
   type Struct struct {
     // fields...
   }
   ```

oliverpool committed:
dedicated Struct struct

oliverpool commented:
Alright, I reverted the changes to `Methods` and added a dedicated `Struct` type: 

![2023-01-17-101339](https://user-images.githubusercontent.com/3864879/212857208-388f12ba-b42b-450d-9344-8ef3b6509a17.png)

fluhus commented:
Fantastic!
